docs/HowToGuides/GettingStarted.md

```diff
@@ -253,41 +253,30 @@ Phew, that's a lot to digest! Now let's proceed to the actual build itself!
 
 ### The actual build
 
-1. Build the toolchain with optimizations, debuginfo, and assertions, using
-   Ninja.
-   - macOS:
-     ```sh
-     utils/build-script --skip-build-benchmarks \
-       --skip-ios --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
-       --sccache --release-debuginfo --swift-disable-dead-stripping
-     ```
-   - Linux:
-     ```sh
-     utils/build-script --release-debuginfo
-     ```
-     If you installed and want to use Sccache, include the `--sccache` option in
-     the invocation as well.
-   <!-- FIXME: Without this "hard" line break, the note doesn’t get properly spaced from the bullet -->
-   <br />
+Build the toolchain with optimizations, debuginfo, and assertions, using Ninja:
 
-   > **Note**  
-   > If you are planning to work on the compiler on macOS, but not the parts that are
-   > written in Swift, pass `--bootstrapping=hosttools` to speed up local
-   > development.
+- macOS:
+  ```sh
+  utils/build-script --skip-build-benchmarks \
+    --skip-ios --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
+    --sccache --release-debuginfo --swift-disable-dead-stripping \
+    --bootstrapping=hosttools
+  ```
+- Linux:
+  ```sh
+  utils/build-script --release-debuginfo
+  ```
+  - If you want to additionally build the Swift core libraries, i.e.,
+    swift-corelibs-libdispatch, swift-corelibs-foundation, and
+    swift-corelibs-xctest, add `--xctest` to the invocation.
 
-   This will create a directory `swift-project/build/Ninja-RelWithDebInfoAssert`
-   containing the Swift compiler and standard library and clang/LLVM build artifacts.
-   If the build fails, see [Troubleshooting build issues](#troubleshooting-build-issues).
+- If you installed and want to use Sccache, add `--sccache` to the invocation.
+- If you want to use a debugger such as LLDB on compiler sources, add
+  `--debug-swift` to the invocation.
```

Comment:
You want to replace `--release-debuginfo` with `--debug-swift`, not just add it. Also, the previous note explained why, might be good to keep some of that.

Reply:
To my knowledge, replacing it is equivalent to not specifying the build variant at all, since the default build variant is `--debug`. My experience as a contributor tells me that a newcomer that actually needs a full debug build is an exception. I am yet to feel in need of e.g. stepping over LLVM code when working on the compiler myself.

> Also, the previous note explained why, might be good to keep some of that

👍

Reply:
I'm looking into what is set on linux with the different commands, will let you know what I find.

Reply:
You are right: `--release-debuginfo` sets that for all repos, while `--debug-swift` successfully overrides that for just the Swift compiler frontend, so you would use both. I was worried that they would not interact well, but they do.

Reply:
Also in my experience I've actually always specified `--release` and `--debug-swift` because I am on a machine with `128GB` disk space only so, couldn't have that cost of a `--release-debuginfo` which is not needed for most of contributors working only on the swift target (although may be useful when you see a crash in an LLVM method for example). @AnthonyLatsis, although this topic is out of scope for this PR do you think is worth mentioning that somewhere here?

Reply:
@LucianoPAlmeida Somehow I missed this, sorry. I do think that all these shenanigans that help you optimize build times and save disk space deserve a detailed discussion, but in a more dedicated place like the *DevelopmentTips* document. Seeing as we are moving toward replacing this content with a ready-to-go first-time-contributor-tuned preset and keeping things really simple, we have to make a choice. LLVM assertion failures are a common kind of crash and effect of playing around with unfamiliar code, and beginners make lots of mistakes, so I think the debug info is generally a justified asset for the aspiring contributor.

We absolutely should pursue other ways of improving this axis of experience, though. For example, optimize the checkout, skip non-essential targets, and provide more fine-grained configuration options.

Reply:
> We absolutely should pursue other ways of improving this axis of experience, though. For example, optimize the checkout, skip non-essential targets, and provide more fine-grained configuration options.

Sounds good @AnthonyLatsis :)
I think my point about optimize build times and save disk space is that it is important specially for people who want to contribute but are not on high-end machines, so that helps the experience overall for them.